src/components/content-tab-assistant.tsx

```diff
@@ -173,35 +214,39 @@ export function ContentTabAssistant( { selectedSite }: ContentTabAssistantProps
 	const { __ } = useI18n();
 	const lastMessage = messages.length === 0 ? undefined : messages[ messages.length - 1 ];
 
+	const [ errorMessageId, setErrorMessageId ] = useState< number | null >( null );
+	const [ isThinking, setIsThinking ] = useState( false );
+
 	useEffect( () => {
 		fetchWelcomeMessages();
 	}, [ fetchWelcomeMessages, selectedSite ] );
 
-	const handleSend = async ( messageToSend?: string ) => {
+	const handleSend = async ( messageToSend?: string, messageId?: number ) => {
 		const chatMessage = messageToSend || input;
 		if ( chatMessage.trim() ) {
-			addMessage( chatMessage, 'user', chatId );
+			const newMessageId = messageId ?? messages.length;
+			if ( messageId === undefined ) {
+				addMessage( chatMessage, 'user', chatId );
+			}
 			setInput( '' );
+			setIsThinking( true );
+			setErrorMessageId( null );
 			try {
 				const { message, chatId: fetchedChatId } = await fetchAssistant(
 					chatId,
-					[ ...messages, { content: chatMessage, role: 'user', createdAt: Date.now() } ],
+					[
+						...messages,
+						{ content: chatMessage, role: 'user', createdAt: Date.now(), id: newMessageId },
```

Comment:
When retrying a message we'll append the same message to the AI. This might have side effects on the response, so I'd propose just sending the same messages before the error.